rs/nns/integration_tests/src/neuron_following.rs

```diff
@@ -537,6 +539,148 @@ async fn test_prune_some_following() {
     }
 }
 
+#[tokio::test]
+async fn test_backfill_voting_power_refreshed_timestamps() {
+    // Step 1: Prepare the world. (This mainly consists of initializing NNS
+    // governance canister with some specially crafted neurons.)
+
+    let pocket_ic = PocketIcBuilder::new().with_nns_subnet().build_async().await;
+
+    let now_seconds = pocket_ic
+        .get_time()
+        .await
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    // This is used in dissolve_state to avoid making the neuron "active" (for
+    // the purposes of storage location). To control whether a neuron is active,
+    // we instead use the cached_neuron_stake_e8s field.
+    let earlier_timestamp_seconds = now_seconds - ONE_MONTH_SECONDS;
+
+    let neuron_base = Neuron {
+        dissolve_state: Some(DissolveState::WhenDissolvedTimestampSeconds(
+            earlier_timestamp_seconds,
+        )),
+        created_timestamp_seconds: earlier_timestamp_seconds,
+        aging_since_timestamp_seconds: u64::MAX,
+
+        // This is so that original_neurons will match what gets
+        // returned by the governance canister.
+        visibility: Some(Visibility::Private as i32),
+
+        ..Default::default()
+    };
+
+    // When this occurs in the voting_power_refreshed_timestamp_seconds field,
+    // that field is supposed to get set to (the new)
+    // DEFAULT_VOTING_POWER_REFRESHED_TIMESTAMP_SECONDS.
+    const EVIL_TIMESTAMP_SECONDS: u64 = 1731628801;
+    // Assert that evil timestamp is "much greater" than the proper value.
+    {
+        let seconds = EVIL_TIMESTAMP_SECONDS - DEFAULT_VOTING_POWER_REFRESHED_TIMESTAMP_SECONDS;
+        assert!(
+            seconds as f64 / ONE_DAY_SECONDS as f64 > 60.0,
+            "{} vs. {}",
+            EVIL_TIMESTAMP_SECONDS as f64 / ONE_DAY_SECONDS as f64,
+            DEFAULT_VOTING_POWER_REFRESHED_TIMESTAMP_SECONDS as f64 / ONE_DAY_SECONDS as f64,
+        );
+    }
+
+    let voting_power_refreshed_timestamp_seconds_neuron_values = vec![
+        EVIL_TIMESTAMP_SECONDS - 1,
+        EVIL_TIMESTAMP_SECONDS,
+        EVIL_TIMESTAMP_SECONDS + 1,
+        now_seconds,
+    ];
+
+    let cached_neuron_stake_e8s_neuron_values = vec![
+        0,       // inactive -> stable memory
+        42 * E8, // active -> heap
+    ];
+
+    let original_neurons = voting_power_refreshed_timestamp_seconds_neuron_values
+        .into_iter()
+        .cartesian_product(cached_neuron_stake_e8s_neuron_values.into_iter())
+        .collect::<Vec<_>>();
+
+    let controller = PrincipalId::new_user_test_id(482_783_461);
+
+    let ids = 100..;
+    let original_neurons = ids
+        .zip(original_neurons.into_iter())
```

Comment:
NCR;

Interesting. I typically use `.enumerate()` to achieve similar things. But I can't see why this is an inferior approach.